### ISSUE TITLE
Kubernetes just compiles on amd64

### ISSUE THREAD

**Issue by luxas** [kubernetes/kubernetes#17060]
Hello,
Since you vendored docker libcontainer, a bug also came along: https://github.com/docker/docker/issues/15053

Very easy to fix indeed: 
```
mv Godeps/_workspace/src/github.com/docker/libcontainer/seccomp/jump{_amd64,}.go
sed -e "s@,amd64@@" -i Godeps/_workspace/src/github.com/docker/libcontainer/seccomp/jump.go
```

It doesn't work on ARM at least.
Is there anything we can do to support out-of-the-box other platforms also?

**Comment by paralin** [comment]
I think cross-compilation works, but compilation *on* those platforms does not.

**Comment by luxas** [comment]
That sounds logical. 
However, is it possible to cross-compile dynamically? 
E. g. hyperkube requires that.

**Comment by yorele** [comment]
related question: i'm confused about compiling for arm: if i can compile the Go code, the release is made in Docker images, and as far as I can tell they use standard Centos and Debian:jessie base images so they are certainly not compatible with arm out of the box. In order to be compatible with arm32 the base image would have to be a debian armhf or centos arm32 image.

considering this, i was going to compile for arm and then build specific arm Docker images, or am I missing something here?
problem I see right now is that hypercube doesn't get built for linux/arm for example, and it's one of the key binary. Is this related to the comment from @luxas ?

thanks for your feedback on this.

**Comment by paralin** [comment]
@yorele You'll need to actually use arm base images. 

See https://github.com/paralin/crew/blob/master/plugins/scratch/commands

It's a system that will automatically scratch re-compile images.

However, in this case you probably want to just modify the base images to armbuild/debian

There's no really good automatic solution at the moment, I believe.

**Comment by luxas** [comment]
Yeah, @yorele you'll need to use ARM base images, if you **compile** on ARM.
But, AFAIK you may cross-compile **statically** to ARM, but not dynamically.

See my project here: https://github.com/luxas/kubernetes-on-arm
Prebuilt k8s binaries on ARM: https://github.com/luxas/kubernetes-on-arm/releases

**Comment by luxas** [comment]
This issue may be closed in favor for #17981
